moto/events/models.py

```diff
@@ -200,6 +210,146 @@ def delete_from_cloudformation_json(
+        if description:
+            self.description = description
+        if event_pattern:
+            self.event_pattern = event_pattern
+        if retention:
+            self.retention = retention
+
+    def delete(self, region_name):
+        event_backend = events_backends[region_name]
+        event_backend.archives.pop(self.name)
+
+    def matches_pattern(self, event):
+        if not self.event_pattern:
+            return True
+
+        # only works on the first level of the event dict
+        # logic for nested dicts needs to be implemented
+        for pattern_key, pattern_value in json.loads(self.event_pattern).items():
+            event_value = event.get(pattern_key)
+            if event_value not in pattern_value:
+                return False
+
+    def get_cfn_attribute(self, attribute_name):
+        from moto.cloudformation.exceptions import UnformattedGetAttTemplateException
+
+        if attribute_name == "ArchiveName":
+            return self.name
+        elif attribute_name == "Arn":
+            return self.arn
+
+        raise UnformattedGetAttTemplateException()
+
+    @staticmethod
+    def cloudformation_name_type():
+        return "ArchiveName"
+
+    @staticmethod
+    def cloudformation_type():
+        # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-events-archive.html
+        return "AWS::Events::Archive"
+
+    @classmethod
+    def create_from_cloudformation_json(
+        cls, resource_name, cloudformation_json, region_name
+    ):
+        properties = cloudformation_json["Properties"]
+        event_backend = events_backends[region_name]
+
+        source_arn = properties.get("SourceArn")
+        description = properties.get("Description")
+        event_pattern = properties.get("EventPattern")
+        retention = properties.get("RetentionDays")
+
+        return event_backend.create_archive(
+            resource_name, source_arn, description, event_pattern, retention
+        )
+
+    @classmethod
+    def update_from_cloudformation_json(
+        cls, original_resource, new_resource_name, cloudformation_json, region_name
+    ):
+        if new_resource_name == original_resource.name:
+            properties = cloudformation_json["Properties"]
+
+            original_resource.update(
+                properties.get("Description"),
+                properties.get("EventPattern"),
+                properties.get("Retention"),
+            )
+
+            return original_resource
+        else:
+            original_resource.delete(region_name)
+            return cls.create_from_cloudformation_json(
+                new_resource_name, cloudformation_json, region_name
+            )
+
+    @classmethod
+    def delete_from_cloudformation_json(
+        cls, resource_name, cloudformation_json, region_name
+    ):
+        event_backend = events_backends[region_name]
+        event_backend.delete_event_bus(resource_name)
```

Comment:
Thanks, I will take a look.

Reply:
ok now I know, why it works, actually it is an interesting behaviour. If the Class (here Archive) has a method `delete()` implemented, it will be invoked by `ResourceMap`
https://github.com/spulec/moto/blob/651998853b0471d5e382e8ebdaa97551b289730d/moto/cloudformation/parsing.py#L685-L686

Reply:
so it is actually dead code, which will not be used.